merky/test/exclude_annotation_nesting_test.py

```diff
@@ -0,0 +1,153 @@
+import collections
+
+from nose import tools
+from . import words
+import merky
+
+BASIC_SEQ = (words.SHEKELS, words.EUROS, words.EPEES)
+BASIC_SEQ_JSON = words.unify('[',
+                                '"', words.SHEKELS, '",',
+                                '"', words.EUROS, '",',
+                                '"', words.EPEES, '"',
+                             ']')
+BASIC_SEQ_HASH = '8129cb64b06e72be46eec5037eafc1583586325a'
+
+UNANNOTATED_NESTED_SEQ = (words.ANGSTROM, BASIC_SEQ, 213, -315)
+UNANNOTATED_NESTED_SEQ_JSON = words.unify('[',
+                                            '"', words.ANGSTROM, '",',
+                                            BASIC_SEQ_JSON, ',',
+                                            '213,',
+                                            '-315',
+                                          ']')
+UNANNOTATED_NESTED_SEQ_HASH = '14fe00c4e749eb96b8c66370902ac90a4f0d52fe'
+
+BASIC_DICT = {words.SHEKELS: words.EUROS, words.ANGSTROM: words.EPEES}
+BASIC_DICT_ORDERED = collections.OrderedDict((
+                        (words.ANGSTROM, words.EPEES),
+                        (words.SHEKELS, words.EUROS),
+                     ))
+BASIC_DICT_JSON = words.unify('{',
+                                '"', words.ANGSTROM, '":"', words.EPEES, '",',
+                                '"', words.SHEKELS, '":"', words.EUROS, '"',
+                              '}')
+BASIC_DICT_HASH = 'b22023acbbc3af979ba46d986706d8da7614f722'
+
+UNANNOTATED_NESTED_DICT = {"dict": dict(BASIC_DICT), "list": BASIC_SEQ}
+UNANNOTATED_NESTED_DICT_ORDERED = collections.OrderedDict((
+                                    ("dict", BASIC_DICT_ORDERED),
+                                    ("list", list(BASIC_SEQ))
+                                  ))
+UNANNOTATED_NESTED_DICT_JSON = words.unify('{',
+                                            '"dict":', BASIC_DICT_JSON, ',',
+                                            '"list":', BASIC_SEQ_JSON,
+                                           '}')
+UNANNOTATED_NESTED_DICT_HASH = '616bcc08fa64f4e8ed9cd7395c8981cdff045baf'
+UNANNOTATED_NESTED_DICT_TOKENIZED = collections.OrderedDict((
+                                      ("dict", BASIC_DICT_HASH),
+                                      ("list", BASIC_SEQ_HASH)
+                                     ))
+
+
+
+
+def test_simple_unannotated_list():
+    # It always does the top level guy regardless of annotation.
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(BASIC_SEQ)
+    tools.assert_equal((BASIC_SEQ_HASH, list(BASIC_SEQ)), next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_nested_unannotated_list():
+    # Top level always gets done, but this time, the nested guy wil be too.
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(UNANNOTATED_NESTED_SEQ)
+    # We get the inner guy first, who is tokenized due to annotation
```

Comment:
I think this particular comment is a copy-paste relic.  You get the inner guy first because it ISN'T annotated.